api_app/analyzers_manager/file_analyzers/malprob.py

```diff
@@ -0,0 +1,78 @@
+import logging
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class MalprobScan(FileAnalyzer):
+    url: str = "https://malprob.io/api"
+    rescan: bool = False
+    _api_key_name: str
+
+    def update(self):
+        pass
+
+    def run(self):
+        response = {}
+        file_name = str(self.filename).replace("/", "_").replace(" ", "_")
+        headers = {"Authorization": f"Token {self._api_key_name}"}
+        binary_file = self.read_file_bytes()
+
+        if not self.rescan:
+            logger.info(f"uploading {file_name} to MalProb.io for analysis")
+            response["scan"] = requests.post(
+                f"{self.url}/scan/",
+                files={"file": binary_file},
+                data={"name": file_name, "private": False},
+                headers=headers,
+                timeout=120,
+            )
+            response["scan"].raise_for_status()
+            if response["scan"].status_code == 204:
+                raise AnalyzerRunException("Limit reached for API")
+            elif response["scan"].status_code == 302:
+                raise logger.error(
```

Comment:
raise not necessary + I would use an info log cause that's something that could normally happen

Reply:
and add md5 in the log